src/librustc_mir/borrow_check/nll/type_check/mod.rs

```diff
@@ -1910,6 +1911,25 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
                         }
                     }
 
+                    CastKind::Hide => {
+                        let predicates = match ty.sty {
+                            ty::Opaque(def_id, substs) => {
+                                let bounds = tcx.predicates_of(def_id);
+                                let result = bounds.instantiate(tcx, substs);
```

Comment:
use this function instead

```rust
 fn normalize_and_prove_instantiated_predicates(
        &mut self,
        instantiated_predicates: ty::InstantiatedPredicates<'tcx>,
        locations: Locations,
    ) {
```

(it's elsewhere in the file)

Reply:
Okay looks good.